File: Practica5/Marco teorico/3. Destructor/3.cpp
#include <iostream>
#include <string>


using namespace std;


class Persona {
private:
    string nombre;


public:
    // Constructor
    Persona(string n) : nombre(n) {
        cout << "Constructor llamado para " << nombre << endl;
    }


    // Destructor
    ~Persona() {
        cout << "Destructor llamado para " << nombre << endl;
    }


    // Método para mostrar el nombre de la persona
    void mostrarNombre() {
        cout << "Nombre: " << nombre << endl;
    }
};


int main() {
    // Crear un objeto Persona
    Persona* persona1 = new Persona("Juan");


    // Llamar al método para mostrar el nombre de la persona
    persona1->mostrarNombre();


    // Liberar la memoria asignada al objeto persona1
    delete persona1;


    return 0;
}

File: Practica5/Marco teorico/3. Destructor/3.java
public class Persona {
    private String nombre;

    // Constructor
    public Persona(String n) {
        nombre = n;
        System.out.println("Constructor llamado para " + nombre);
    }

    // Método para mostrar el nombre de la persona
    public void mostrarNombre() {
        System.out.println("Nombre: " + nombre);
    }

    // Destructor (no es necesario en Java)
}

public class Main {
    public static void main(String[] args) {
        // Crear un objeto Persona
        Persona persona1 = new Persona("Juan");

        // Llamar al método para mostrar el nombre de la persona
        persona1.mostrarNombre();

        // No es necesario liberar la memoria en Java (lo hace el recolector de basura)
    }
}

